diff --git a/integration-test-common/docker-script/init/trino/config/jvm.config b/integration-test-common/docker-script/init/trino/config/jvm.config
index 77a930a36..e1089ee4b 100644
--- a/integration-test-common/docker-script/init/trino/config/jvm.config
+++ b/integration-test-common/docker-script/init/trino/config/jvm.config
@@ -26,5 +26,8 @@
 -XX:+ExplicitGCInvokesConcurrent
 -XX:+HeapDumpOnOutOfMemoryError
 
+-XX:+UnlockDiagnosticVMOptions
+-XX:G1NumCollectionsKeepPinned=10000000
+
 -DHADOOP_USER_NAME=hive
 -Dlog4j.configurationFile=/etc/trino/log4j2.properties
diff --git a/trino-connector/integration-test/src/test/resources/trino-ci-testset/testsets/lakehouse-iceberg/00011_table_statistics.sql b/trino-connector/integration-test/src/test/resources/trino-ci-testset/testsets/lakehouse-iceberg/00011_table_statistics.sql
new file mode 100644
index 000000000..a3154a68e
--- /dev/null
+++ b/trino-connector/integration-test/src/test/resources/trino-ci-testset/testsets/lakehouse-iceberg/00011_table_statistics.sql
@@ -0,0 +1,14 @@
+CREATE SCHEMA gt_iceberg_test_table_statistics;
+
+CREATE TABLE gt_iceberg_test_table_statistics.tb01 (
+    key1 int,
+    value1 int
+);
+
+INSERT INTO gt_iceberg_test_table_statistics.tb01(key1, value1) VALUES (1, 1), (2, 2);
+
+SHOW STATS FOR gt_iceberg_test_table_statistics.tb01;
+
+DROP TABLE gt_iceberg_test_table_statistics.tb01;
+
+DROP SCHEMA gt_iceberg_test_table_statistics;
diff --git a/trino-connector/integration-test/src/test/resources/trino-ci-testset/testsets/lakehouse-iceberg/00011_table_statistics.txt b/trino-connector/integration-test/src/test/resources/trino-ci-testset/testsets/lakehouse-iceberg/00011_table_statistics.txt
new file mode 100644
index 000000000..ce9f3101b
--- /dev/null
+++ b/trino-connector/integration-test/src/test/resources/trino-ci-testset/testsets/lakehouse-iceberg/00011_table_statistics.txt
@@ -0,0 +1,13 @@
+CREATE SCHEMA
+
+CREATE TABLE
+
+INSERT: 2 rows
+
+"key1","","","0.0","","1","2"
+"value1","","","0.0","","1","2"
+"","","","","2.0","",""
+
+DROP TABLE
+
+DROP SCHEMA
diff --git a/trino-connector/integration-test/src/test/resources/trino-ci-testset/testsets/lakehouse-iceberg/catalog_iceberg_cleanup.sql b/trino-connector/integration-test/src/test/resources/trino-ci-testset/testsets/lakehouse-iceberg/catalog_iceberg_cleanup.sql
new file mode 100644
index 000000000..051d2a503
--- /dev/null
+++ b/trino-connector/integration-test/src/test/resources/trino-ci-testset/testsets/lakehouse-iceberg/catalog_iceberg_cleanup.sql
@@ -0,0 +1 @@
+CALL gravitino.system.drop_catalog('gt_iceberg');
diff --git a/trino-connector/integration-test/src/test/resources/trino-ci-testset/testsets/lakehouse-iceberg/catalog_iceberg_mysql_cleanup.sql b/trino-connector/integration-test/src/test/resources/trino-ci-testset/testsets/lakehouse-iceberg/catalog_iceberg_mysql_cleanup.sql
new file mode 100644
index 000000000..19d744acd
--- /dev/null
+++ b/trino-connector/integration-test/src/test/resources/trino-ci-testset/testsets/lakehouse-iceberg/catalog_iceberg_mysql_cleanup.sql
@@ -0,0 +1 @@
+CALL gravitino.system.drop_catalog('gt_iceberg_mysql');
diff --git a/trino-connector/integration-test/src/test/resources/trino-ci-testset/testsets/lakehouse-iceberg/catalog_iceberg_mysql_prepare.sql b/trino-connector/integration-test/src/test/resources/trino-ci-testset/testsets/lakehouse-iceberg/catalog_iceberg_mysql_prepare.sql
new file mode 100644
index 000000000..607be5081
--- /dev/null
+++ b/trino-connector/integration-test/src/test/resources/trino-ci-testset/testsets/lakehouse-iceberg/catalog_iceberg_mysql_prepare.sql
@@ -0,0 +1,17 @@
+call gravitino.system.create_catalog(
+    'gt_iceberg_mysql',
+    'lakehouse-iceberg',
+    map(
+        array['uri', 'catalog-backend', 'warehouse', 'jdbc-user', 'jdbc-password', 'jdbc-driver'],
+        array['${mysql_uri}/iceberg_db?createDatabaseIfNotExist=true&useSSL=false', 'jdbc',
+            '${hdfs_uri}/user/iceberg/warehouse/TrinoQueryIT', 'trino', 'ds123', 'com.mysql.cj.jdbc.Driver']
+    )
+);
+
+show catalogs;
+
+CREATE SCHEMA gt_iceberg_mysql.gt_db2;
+
+USE gt_iceberg_mysql.gt_db2;
+
+DROP SCHEMA gt_iceberg_mysql.gt_db2;
diff --git a/trino-connector/integration-test/src/test/resources/trino-ci-testset/testsets/lakehouse-iceberg/catalog_iceberg_prepare.sql b/trino-connector/integration-test/src/test/resources/trino-ci-testset/testsets/lakehouse-iceberg/catalog_iceberg_prepare.sql
new file mode 100644
index 000000000..46ad6a38c
--- /dev/null
+++ b/trino-connector/integration-test/src/test/resources/trino-ci-testset/testsets/lakehouse-iceberg/catalog_iceberg_prepare.sql
@@ -0,0 +1,16 @@
+call gravitino.system.create_catalog(
+    'gt_iceberg',
+    'lakehouse-iceberg',
+    map(
+        array['uri', 'catalog-backend', 'warehouse'],
+        array['${hive_uri}', 'hive', '${hdfs_uri}/user/iceberg/warehouse/TrinoQueryIT']
+    )
+);
+
+show catalogs;
+
+CREATE SCHEMA gt_iceberg.gt_db2;
+
+USE gt_iceberg.gt_db2;
+
+DROP SCHEMA gt_iceberg.gt_db2;
diff --git a/trino-connector/integration-test/src/test/resources/trino-ci-testset/testsets/lakehouse-iceberg/ignored/00011_table_statistics.sql b/trino-connector/integration-test/src/test/resources/trino-ci-testset/testsets/lakehouse-iceberg/ignored/00011_table_statistics.sql
deleted file mode 100644
index a3154a68e..000000000
--- a/trino-connector/integration-test/src/test/resources/trino-ci-testset/testsets/lakehouse-iceberg/ignored/00011_table_statistics.sql
+++ /dev/null
@@ -1,14 +0,0 @@
-CREATE SCHEMA gt_iceberg_test_table_statistics;
-
-CREATE TABLE gt_iceberg_test_table_statistics.tb01 (
-    key1 int,
-    value1 int
-);
-
-INSERT INTO gt_iceberg_test_table_statistics.tb01(key1, value1) VALUES (1, 1), (2, 2);
-
-SHOW STATS FOR gt_iceberg_test_table_statistics.tb01;
-
-DROP TABLE gt_iceberg_test_table_statistics.tb01;
-
-DROP SCHEMA gt_iceberg_test_table_statistics;
diff --git a/trino-connector/integration-test/src/test/resources/trino-ci-testset/testsets/lakehouse-iceberg/ignored/00011_table_statistics.txt b/trino-connector/integration-test/src/test/resources/trino-ci-testset/testsets/lakehouse-iceberg/ignored/00011_table_statistics.txt
deleted file mode 100644
index ce9f3101b..000000000
--- a/trino-connector/integration-test/src/test/resources/trino-ci-testset/testsets/lakehouse-iceberg/ignored/00011_table_statistics.txt
+++ /dev/null
@@ -1,13 +0,0 @@
-CREATE SCHEMA
-
-CREATE TABLE
-
-INSERT: 2 rows
-
-"key1","","","0.0","","1","2"
-"value1","","","0.0","","1","2"
-"","","","","2.0","",""
-
-DROP TABLE
-
-DROP SCHEMA
diff --git a/trino-connector/integration-test/src/test/resources/trino-ci-testset/testsets/lakehouse-iceberg/ignored/catalog_iceberg_mysql_cleanup.sql b/trino-connector/integration-test/src/test/resources/trino-ci-testset/testsets/lakehouse-iceberg/ignored/catalog_iceberg_mysql_cleanup.sql
deleted file mode 100644
index 19d744acd..000000000
--- a/trino-connector/integration-test/src/test/resources/trino-ci-testset/testsets/lakehouse-iceberg/ignored/catalog_iceberg_mysql_cleanup.sql
+++ /dev/null
@@ -1 +0,0 @@
-CALL gravitino.system.drop_catalog('gt_iceberg_mysql');
diff --git a/trino-connector/integration-test/src/test/resources/trino-ci-testset/testsets/lakehouse-iceberg/ignored/catalog_iceberg_mysql_prepare.sql b/trino-connector/integration-test/src/test/resources/trino-ci-testset/testsets/lakehouse-iceberg/ignored/catalog_iceberg_mysql_prepare.sql
deleted file mode 100644
index 607be5081..000000000
--- a/trino-connector/integration-test/src/test/resources/trino-ci-testset/testsets/lakehouse-iceberg/ignored/catalog_iceberg_mysql_prepare.sql
+++ /dev/null
@@ -1,17 +0,0 @@
-call gravitino.system.create_catalog(
-    'gt_iceberg_mysql',
-    'lakehouse-iceberg',
-    map(
-        array['uri', 'catalog-backend', 'warehouse', 'jdbc-user', 'jdbc-password', 'jdbc-driver'],
-        array['${mysql_uri}/iceberg_db?createDatabaseIfNotExist=true&useSSL=false', 'jdbc',
-            '${hdfs_uri}/user/iceberg/warehouse/TrinoQueryIT', 'trino', 'ds123', 'com.mysql.cj.jdbc.Driver']
-    )
-);
-
-show catalogs;
-
-CREATE SCHEMA gt_iceberg_mysql.gt_db2;
-
-USE gt_iceberg_mysql.gt_db2;
-
-DROP SCHEMA gt_iceberg_mysql.gt_db2;
